### ISSUE TITLE
WorkflowError: Metadata can't be created as it already exists (Windows)

### ISSUE THREAD

**Issue by DaminK** [snakemake/snakemake#1882]
[Others ](https://github.com/snakemake/snakemake/issues/1788#issuecomment-1221312000) seem to have the same problem on windows, but it was commented on an unrelated Issue so I'll open a seperate issue. 

**Snakemake version:** 7.14.2 (latest stable release)

**Describe the bug**
If a previous job is run again (due to changes in code, config or manually removing output files that are required by a subsequent rule), it sometimes fails to create the corresponding meta data as it is already existing. 

This can be circumvented by either:
- manually removing the metadata folder
- restarting the rule/file again (but likely fails with the same error on a subsequent job that was also already executed, leading to several restarts of a rule/file to get the final results)

**Logs**
```
WorkflowError:
Error recording metadata for finished job ([WinError 183] Cannot create a file when that file already exists: 'D:\\WIPAR\\repo\\.snakemake\\metadata\\tmp6cvr9il8.cmVzdWx0' -> 'D:\\WIPAR\\repo\\.snakemake\\metadata\\cmVzdWx0cy9HTjA2XzIwMjEtMDEtMjBfMTAtMTUtMTYvYW5hdG9taWNhbC9BbGwvRmVhdHVyZXMvdmlzdWFsL2F1dG9jb3ZhcmlhbmNlX3RpbWVsYWdzflsxXS9mZWF0dXJlcy5oNQ=='). Please ensure write permissions for the directory D:\WIPAR\repo\.snakemake
Removing output files of failed job feature_calculation since they might be corrupted:
results/GN06_2021-01-20_10-15-16/anatomical/All/Features/visual/autocovariance_timelags~[1]/features.h5, results/GN06_2021-01-20_10-15-16/anatomical/All/Features/visual/autocovariance_timelags~[1]/features.npy, results/GN06_2021-01-20_10-15-16/anatomical/All/Features/visual/autocovariance_timelags~[1]/conf.yaml
```
The WinError was in the language of my OS (German), so I substituted the English translation.

**Minimal example**
Can be reproduced with any Snakefile. For simplicity I tested it with one of your tests, the [test_many_jobs](https://github.com/snakemake/snakemake/tree/main/tests/test_many_jobs). On windows I changed the `touch {output}` to `type nul > {output}` as touch is not supported in cmd. Now you can execute `snakemake -j4 1.out` to produce the 1.out file with rules 2,3,4,5. If you delete 1.out and call `snakemake -j4 1.out` again you get the above mentionend error. Here is the log of both runs and the slightly modified snakefile: 
1. Run [2022-09-29T170518.672838.snakemake.log](https://github.com/snakemake/snakemake/files/9675938/2022-09-29T170518.672838.snakemake.log)
2. Run with 1.out removed [2022-09-29T170534.139110.snakemake.log](https://github.com/snakemake/snakemake/files/9675937/2022-09-29T170534.139110.snakemake.log)
3. Snakefile (as *.txt so I can upload it here) [Snakefile.txt](https://github.com/snakemake/snakemake/files/9675952/Snakefile.txt)

**Additional context**
Only tested on windows. Only started to happen since last month when I upgraded Snakemake. Unfortunately, I don't know the exact version I had before (likely 3-6 month old). 

Seems to be related to creating symbolic links (`D:\\WIPAR\\....' -> 'D:\\WIPAR\\...`). Some time ago, I had to change the required privileges to create symbolic links in python without requiring admin rights. May be related.

**Comment by Hugovdberg** [comment]
Same problem on version 7.14.0

**Comment by JoerivanEngelen** [comment]
Just adding info here (as I did not see potential fixes in the changelog, so expected): 
I also ran into this issue with version 7.17.1 on Windows.

Thanks for raising this issue, I ran into it quite a few times now and it's quite a frustrating issue with long running rules (e.g. rerun 1 hour of processing, because of a permission error). However, I did not find time to investigate it yet.

I vaguely remember reading an issue here that Snakemake still lacked a test which would run twice, but I'm unsure. I think if the example you made would be converted to a test (by one of us), it would be a good addition to the test bench.

**Comment by Hugovdberg** [comment]
even if the multiple runs of finished causes an error this should be distinguished from an actual error with the contents of the rule. Deleting expensive data is very annoying to say the least

**Comment by sfritzke** [comment]
Same problem on version 7.18.1

**Comment by stanmart** [comment]
The root cause is the platform-specific behavior of `os.rename` in `snakemake/persistence.py` https://github.com/snakemake/snakemake/blob/a5c3523992656ec8230d59ee12f9652313c6dd96/snakemake/persistence.py#L462

Fix proposed in PR #1971

**Comment by stanmart** [comment]
Hey @DaminK, @JoerivanEngelen, @sfritzke, this issue should be fixed in 7.19.0. Could you check if it solves your problems, in which case this can probably be closed?

**Comment by JoerivanEngelen** [comment]
Just tried it quickly, and I can confirm the issue is fixed with 7.19.0! A colleague also had this experience today. Thanks for the fix!
